### ISSUE TITLE
Models sometimes won't be created after sync and close for SQLite3

### ISSUE THREAD

**Issue by Coteh** [eveningkid/denodb#22]
## Reproduction Steps
1. Create two models, don't think it matters what's in them, as long as there's a primary key and at least one other column
1. After that, call `await db.link([Model1, Model2])`, then `await db.sync()`, then `await db.close()`
1. If both tables were created successfully, delete one of them then rerun

## Expected Behaviour
Both tables should be created successfully.

## Actual Behaviour
Sometimes both will be created successfully, but sometimes not all of them will be created.

Let me know if there's a misunderstanding of how these methods work or if there's any issues reproducing it. Thanks.

**Comment by Coteh** [comment]
If you have a bunch of models and only a couple of them are being made, a workaround can be to remove all the models from the array except for the one not being created, rerun and repeat for each model that wasn't created. Sometimes rerunning without modifying the array also creates one of the problematic models, but happens infrequently. I suspect some sort of race condition is happening here?

**Comment by eveningkid** [comment]
Hey James,

This is not normal and you made me very curious about why this is happening.

I know you said it happens with pretty much any model, but could you provide a little example I could run on my end? What database are you using?

If it's about a race condition indeed, this should be easily sorted out.

Please let me know :)

**Comment by eveningkid** [comment]
Hey James,

I still couldn't reproduce the issue, any other input on that?

Thank you

**Comment by Coteh** [comment]
Hey Arnaud, sorry about the late response. My internet was out for a few days, just got a replacement modem yesterday. Seems like all tables are created now, after running another minimal example and my web app. I even dropped one of the tables and reran the app and it created it again. I noticed that you upgraded to version 2 of the sqlite library sometime after I posted this issue, maybe that fixed it.

I'll reopen if it happens again.
